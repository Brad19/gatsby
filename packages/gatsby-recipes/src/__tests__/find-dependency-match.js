const findDependencyMatch = require(`../find-dependency-match`)

describe(`findDependencyMatch`, () => {
  test(`basic matching`, () => {
    const resources = [
      {
        resourceName: `foo`,
        resourceDefinitions: {
          name: `foos`,
        },
      },
      {
        resourceName: `foo`,
        resourceDefinitions: {
          name: `bar`,
        },
      },
      {
        resourceName: `foobar`,
        resourceDefinitions: {
          name: `bar`,
        },
      },
    ]
    const resourceToMatch = {
      dependsOn: [
        {
          resourceName: `foo`,
          name: `bar`,
        },
      ],
    }

    const matches = findDependencyMatch(resources, resourceToMatch)
    expect(matches).toHaveLength(1)
    expect(matches).toMatchSnapshot()
  })
})
