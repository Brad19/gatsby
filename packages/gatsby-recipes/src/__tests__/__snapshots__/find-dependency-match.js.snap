// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findDependencyMatch basic matching 1`] = `
Array [
  Object {
    "resourceDefinitions": Object {
      "name": "bar",
    },
    "resourceName": "foo",
  },
]
`;
